pkg/ebpf/tracer_test.go

```diff
@@ -1646,3 +1650,74 @@ func TestConntrackExpiration(t *testing.T) {
 
 	assert.Nil(t, tr.conntracker.GetTranslationForConn(*conn), "translation should have been deleted")
 }
+
+func TestTCPEstablished(t *testing.T) {
+	tr, err := NewTracer(NewDefaultConfig())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tr.Stop()
+
+	// Warm-up state
+	getConnections(t, tr)
+
+	server := NewTCPServer(func(c net.Conn) {
+		io.Copy(ioutil.Discard, c)
+		c.Close()
+	})
+	doneChan := make(chan struct{})
+	server.Run(doneChan)
+	defer close(doneChan)
+
+	c, err := net.DialTimeout("tcp", server.address, 50*time.Millisecond)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	c.Write([]byte("hello"))
+	defer c.Close()
+
+	connections := getConnections(t, tr)
+	conn, ok := findConnection(c.LocalAddr(), c.RemoteAddr(), connections)
+
+	assert.True(t, ok)
+	assert.Equal(t, uint32(1), conn.LastTCPEstablished)
+
+	// Verify that we report the connection establishment only once
+	connections = getConnections(t, tr)
+	conn, ok = findConnection(c.LocalAddr(), c.RemoteAddr(), connections)
+	assert.True(t, ok)
+	assert.Equal(t, uint32(0), conn.LastTCPEstablished)
```

Comment:
Can we close the conn and make sure the LastTCPClosed also goes to 1 ?

Reply:
Sure, will update it 👍